test/e2e/functional_test.go

```diff
@@ -728,6 +728,22 @@ func (s *FunctionalSuite) TestDataTransformation() {
 		})
 }
 
+func (s *FunctionalSuite) TestMarkDaemonedPodSucceeded() {
+	s.Given().
+		Workflow("@testdata/daemoned-pod-completed.yaml").
+		When().
+		SubmitWorkflow().
+		WaitForWorkflow(1 * time.Minute).
+		Then().
+		ExpectWorkflow(func(t *testing.T, metadata *metav1.ObjectMeta, status *wfv1.WorkflowStatus) {
+			assert.Equal(t, wfv1.WorkflowSucceeded, status.Phase)
+			node := status.Nodes.FindByDisplayName("redis")
+			if assert.NotNil(t, node) {
+				assert.Equal(t, wfv1.NodeSucceeded, node.Phase)
+			}
+		})
+}
```

Comment:
Spent around an hour trying to add a non-E2E test for this, but because the relevant code generates a client set from a REST config (and there's no easy way to mock such a config, although there is an easy way to mock a client set, but using the mock clientset would require code changes), I decided to use an E2E test. After doing so I do believe an E2E test is better for this anyway.